### PR TITLE
initial documentation

### PR DESCRIPTION
**Pull request by yuvalif**
* this is not full documentation - user/developer guide are mainly skeletons
* the user/developer guides are not reflect the current state of the code, and are more future looking after some of the issues are resolved

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
-# k8s-node-net-conf [WIP]
-This is in initial phase of implementation. See ```docs/README.md``` for overall design and functionality.
+# The "Why"
+With hybrid clouds, host networking setup is even becoming more challenging. Different payloads have different networking requirements, and not everything could be satisfied as overlays on top of the main interface of the host (SR-IOV, L2, ...).
+The [CNI](https://github.com/containernetworking/cni) standard enables different solutions for connecting networks on the host with pods. Some of them are [part of the standard](https://github.com/containernetworking/cni), and there are some that support: [OVS bridges](https://github.com/containernetworking/cni), [SR-IOV](https://github.com/containernetworking/cni), and more...
```
[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: to which links?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+When running in "client" mode, it has nothing more to do. When running as a daemon, it will:
+- Detects updates to ```NodeNetConfPolicy``` CRDs applicable for current host, and update ```NodeNetworkState``` CRD if needed
+- Detect an update to the current state of ```NodeNetworkState`` CRD for the host, and see if its desired state needs to be modified
+### Centralized mode
+Very similar to the distributed mode, but in this case, the client od daemon must handle policies and states for all nodes in the system.
+# Getting Started
+Please check out our [User Guide](https://github.com/nmstate/kubernetes-nmstate/tree/master/docs/user-guide.md)
```
[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: we first need the complete deployment yaml. once we have that, we can add it here

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+The [CNI](https://github.com/containernetworking/cni) standard enables different solutions for connecting networks on the host with pods. Some of them are [part of the standard](https://github.com/containernetworking/cni), and there are some that support: [OVS bridges](https://github.com/containernetworking/cni), [SR-IOV](https://github.com/containernetworking/cni), and more...
+However, in all of these cases, the host must have the networks setup before the pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster, with dynamic networking requirements, is a challenge by itself - and this is what this project is addressing.
+# The "How"
+We use [nmstate](https://nmstate.github.io/) to perform state driven network configuration on each host, as well as to report back its current state. 
+The system defines 2 CRDs ([Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)): ```NodeNetworkState``` and ```NodeNetConfPolicy```.
```
Comment (dankenigsberg) [inline README.md]: I think you should point here to Eddy's CRD document

File: README.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
-# k8s-node-net-conf [WIP]
-This is in initial phase of implementation. See ```docs/README.md``` for overall design and functionality.
+# The "Why"
+With hybrid clouds, host networking setup is even becoming more challenging. Different payloads have different networking requirements, and not everything could be satisfied as overlays on top of the main interface of the host (SR-IOV, L2, ...).
```
[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: will keep the term "host-networking" but will change host->node anywhere else

File: README.md (hunk 3 above)
Comment (yuvalif) [reply README.md]: not an upstream doc. will do that once it is in git.

Comment (dankenigsberg) [reply README.md]: I don't understand your reply. https://docs.google.com/document/d/1282BcYjYGIIxQKgMYi3nQodB4ML_gw9BSs5AXB7QUtg/edit is upstream, and it is the thing that defines the CRDs that you mention.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (SchSeba) [reply README.md]: [OVS bridges](https://github.com/kubevirt/ovs-cni) 
[SR-IOV](https://github.com/hustcat/sriov-cni)

Maybe change to this?

[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: ok (copy&paste error...)

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: added, assuming the deployment yaml is there

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yuvalif) [reply README.md]: could not fix the 80 char issue. when i add line breaks they are reflected in the actual output

File: README.md (hunk 3 above)
Comment (yuvalif) [reply README.md]: added the reference

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dankenigsberg) [reply README.md]: that's not what
I'm familiar with.
but whatever.

**Pull request merged**